ml-agents/mlagents/trainers/policy/tf_policy.py

```diff
@@ -270,8 +270,16 @@ def get_action(
         )
 
         self.save_memories(global_agent_ids, run_out.get("memory_out"))
+        action = run_out.get("action")
+        # Fast NaN check on the action
+        # See https://stackoverflow.com/questions/6736590/fast-check-for-nan-in-numpy for background.
+        d = np.sum(action)
+        has_nan = np.isnan(d)
+        if has_nan:
+            raise RuntimeError("NaN action detected.")
```

Comment:
Just raising for now; there's not much the user can do about it. We could use `np.nan_to_num` but that would probably trigger every step and give bad results.